### ISSUE TITLE
if Task has a private constructor, you get a NPE

### ISSUE THREAD

**Issue by SiKing** [serenity-bdd/serenity-core#2115]
The other day I created a Task with a private constructor:

```java
public class NavigateToPage implements Task {

    private String label;

    // HERE IS THE PROBLEM:
    private NavigateToPage(String label) {
	this.label = label;
    }

    @Step("{0} navigates to page '#label'")
    @Override
    public <T extends Actor> void performAs(T actor) {

	actor.should(seeThat(the(Page.labeled(label)), isCurrentlyVisible()).orComplainWith(NavigationPageNotFound.class, label));
	actor.attemptsTo(Click.on(Page.labeled(label)));
    }

    public static NavigateToPage labeled(String label) {

	return instrumented(NavigateToPage.class, label);
    }
}
```

This compiled fine. But when I ran this, it gave a rather unhelpful:
```
java.lang.NullPointerException
	at net.serenitybdd.screenplay.InstrumentedTask.isInstrumented(InstrumentedTask.java:69)
	at net.serenitybdd.screenplay.Actor.shouldNotReport(Actor.java:189)
	at net.serenitybdd.screenplay.Actor.attemptsTo(Actor.java:172)
	at net.serenitybdd.screenplay.Actor.wasAbleTo(Actor.java:162)
	at features.cabinet_view.CabinetViewNavigationStory.should_be_able_to_navigate_to_cabinet_view_page(CabinetViewNavigationStory.java:20)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at net.serenitybdd.junit.runners.SerenityStatement.evaluate(SerenityStatement.java:33)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at net.serenitybdd.junit.runners.SerenityRunner.performRunChild(SerenityRunner.java:464)
	at net.serenitybdd.junit.runners.SerenityRunner.runChild(SerenityRunner.java:438)
	at net.serenitybdd.junit.runners.SerenityRunner.runChild(SerenityRunner.java:45)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at net.serenitybdd.junit.runners.SerenityRunner.run(SerenityRunner.java:242)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:89)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:41)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:541)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:763)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:463)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:209)
```

**Comment by wakaleo** [comment]
As you have noticed, tasks cannot have private constructors (because of the way ByteBuddy works). Your PR will produce a better error message.

**Comment by SiKing** [comment]
This was already released, in like 2.2.8?
